routes/attribution.js

```diff
@@ -85,4 +88,74 @@ routes.push({
   },
 });
 
+routes.push({
+  path: prefix(
+    '/{languageCode}/{file}/{typeOfUse}/modified/{modification}/{modificationAuthor}/{licenseId}'
+  ),
+  method: 'GET',
+  options: {
+    description: 'Generate attribution for a modifile work',
+    notes: 'Generate attribution hints for the given file if that file was modified.',
+    validate: {
+      params: {
+        languageCode: Joi.string().valid(knownLanguages),
+        file: Joi.string(),
+        typeOfUse: Joi.string().valid(knownTypesOfUse),
+        modification: Joi.string(),
+        modificationAuthor: Joi.string(),
+        licenseId: Joi.string(),
+      },
+    },
+    response: {
+      schema: attributionSchema,
+      status: {
+        400: definitions.errors['400'],
+        422: definitions.errors['422'],
+        500: definitions.errors['500'],
+        503: definitions.errors['503'],
+      },
+    },
+    plugins: {
+      'hapi-swaggered': {
+        operationId: 'attribution.show',
+        security: [{ default: [] }],
+      },
+    },
+  },
+  handler: async (request, h) => {
+    const { fileData, licenseStore } = request.server.app.services;
+    const {
+      languageCode,
+      file,
+      typeOfUse,
+      modification,
+      modificationAuthor,
+      licenseId,
+    } = request.params;
+    try {
+      const fileInfo = await fileData.getFileData(file);
+      const license = licenseStore.getLicenseById(licenseId);
+      assert.ok(license, errors.licenseNotFound);
```

Comment:
Makes sense to do this. This was actually the bits of "business logic" I thought the attribution service could take care of. Benefit would be to have a less verbose route 🤔

Reply:
I like the idea of extracting this into a separate attribution service 👍 
We could do this as a separate cleanup PR after merging this one.

Reply:
I like the idea (to get the basic app out as soon as possible but to not forget this refactoring). I'll create a ticket.

Reply:
actually I got inspired by @klappradla's use of serializers/factories in another PR and changed my code to use them (and implement the attribution counterparts). This already DRYs the code up a bit.